web/src/components/ak-hint/ak-hint.ts

```diff
@@ -0,0 +1,72 @@
+import { AKElement } from "@goauthentik/app/elements/Base";
+
+import { css, html } from "lit";
+import { customElement } from "lit/decorators.js";
+
+const styles = css`
+    :host {
+        --ak-hint--GridRowGap: var(--pf-global--spacer--md);
+        --ak-hint--PaddingTop: var(--pf-global--spacer--md);
+        --ak-hint--PaddingRight: var(--pf-global--spacer--lg);
+        --ak-hint--PaddingBottom: var(--pf-global--spacer--md);
+        --ak-hint--PaddingLeft: var(--pf-global--spacer--lg);
+        --ak-hint--BackgroundColor: var(--pf-global--palette--blue-50);
+        --ak-hint--BorderColor: var(--pf-global--palette--blue-100);
+        --ak-hint--BorderWidth: var(--pf-global--BorderWidth--sm);
+        --ak-hint--BoxShadow: var(--pf-global--BoxShadow--sm);
+        --ak-hint--Color: var(--pf-global--Color--100);
+
+        /* Hint Title */
+        --ak-hint__title--FontSize: var(--pf-global--FontSize--lg);
+
+        /* Hint Body */
+        --ak-hint__body--FontSize: var(--pf-global--FontSize--md);
+
+        /* Hint Footer */
+        --ak-hint__footer--child--MarginRight: var(--pf-global--spacer--md);
+
+        /* Hint Actions */
+        --ak-hint__actions--MarginLeft: var(--pf-global--spacer--2xl);
+        --ak-hint__actions--c-dropdown--MarginTop: calc(
+            var(--pf-global--spacer--form-element) * -1
+        );
+    }
+
+    :host([theme="dark"]) {
```

Comment:
Once all the sub-components were receiving their styling instructions from this block, adding the host-with-attribute-match made it remarkably easy to get dark mode working.